OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -34,7 +34,7 @@ public Attack(Actor self, Target target, bool allowMovement, bool forceAttack)
 
 			this.forceAttack = forceAttack;
 
-			attack = self.Trait<AttackBase>();
+			attack = self.TraitsImplementing<AttackBase>().First(x => !x.IsTraitDisabled);
```

Comment:
If you store the IEnumerable<AttackBase> then the filtering each tick will be relatively cheap.